powershell-analyzer/src/analyze.ps1

```diff
@@ -0,0 +1,238 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+#Requires -Version 7
+
+<#
+.DESCRIPTION
+    This script runs PSScriptAnalyzer on an entire directory structure, and
+    outputs the results as a Sarif file. This file can be read directly (it is
+    in JSON), or opened in Visual Studio Code with the sarif extension, or it
+    can be uploaded into GitHub Code QL results.
+#>
+param (
+    # Directory in which to do a recursive scan of PowerShell files
+    [Parameter(Mandatory = $True)]
+    [string]
+    $Directory,
+
+    # If set to $false, it will only print the results.
+    [boolean]
+    $SaveToFile = $true,
+
+    # Location of the results file
+    [string]
+    $ResultsPath = "results.sarif",
+
+    # List of excluded rules
+    [string[]]
+    $ExcludedRules = ""
```

Comment:
The analyzer thinks this is an unused parameter. I see that it is used on line 162, inside of a function. Having a parameter outside the function be captured inside the function is an example of closure, and it can be confusing. Suggestion: duplicate the relevant parameters into the function signature for `Invoke-Analyzer`, and then explicitly pass these "global" parameters to the function call on line 230.